### PR TITLE
[Profiler] Update managed projects to net462

### PR DESCRIPTION
**Pull request by gleocadie**
## Summary of changes

Remove net45 framework and replace by net462 for demo apps.

## Reason for change

Too old :)

### CODE REVIEW

**Comment by andrewlock** [comment]
## Execution-Time Benchmarks Report :stopwatch:

Execution-time results for samples comparing the following branches/commits:
- [This PR (4683)](https://github.com/DataDog/dd-trace-dotnet/tree/2dd82abe71860241812ddd948cb3ac35935f6669)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/2b3bf33c6842434ccfdf0ee88e78bd56193fd7c3)

Execution-time benchmarks measure the whole time it takes to execute a program. And are intended to measure the one-off costs. Cases where the execution time results for the PR are worse than latest master results are shown in **red**. The following thresholds were used for comparing the execution times:
* Welch test with statistical test for significance of **5%**
* Only results indicating a difference greater than **5%** and **5 ms** are considered.


Note that these results are based on a _single_ point-in-time result for each branch. For full results, see the [dashboard](https://ddstaging.datadoghq.com/dashboard/4qn-6fi-54p/apm-net-execution-time-benchmarks).

Graphs show the p99 interval based on the mean and StdDev of the test run, as well as the mean value of the run (shown as a diamond below the graph).
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4683) - mean (74ms)  : 58, 91
     .   : milestone, 74,
    master - mean (70ms)  : 61, 79
     .   : milestone, 70,

    section CallTarget+Inlining+NGEN
    This PR (4683) - mean (1,040ms)  : 1015, 1065
     .   : milestone, 1040,
    master - mean (1,037ms)  : 1013, 1062
     .   : milestone, 1037,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4683) - mean (107ms)  : 100, 114
     .   : milestone, 107,
    master - mean (107ms)  : 103, 111
     .   : milestone, 107,

    section CallTarget+Inlining+NGEN
    This PR (4683) - mean (748ms)  : 724, 771
     .   : milestone, 748,
    master - mean (750ms)  : 726, 774
     .   : milestone, 750,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4683) - mean (90ms)  : 87, 93
     .   : milestone, 90,
    master - mean (91ms)  : 85, 97
     .   : milestone, 91,

    section CallTarget+Inlining+NGEN
    This PR (4683) - mean (710ms)  : 683, 736
     .   : milestone, 710,
    master - mean (711ms)  : 691, 732
     .   : milestone, 711,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4683) - mean (190ms)  : 185, 195
     .   : milestone, 190,
    master - mean (191ms)  : 187, 195
     .   : milestone, 191,

    section CallTarget+Inlining+NGEN
    This PR (4683) - mean (1,127ms)  : 1091, 1162
     .   : milestone, 1127,
    master - mean (1,145ms)  : 1118, 1171
     .   : milestone, 1145,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4683) - mean (276ms)  : 271, 282
     .   : milestone, 276,
    master - mean (277ms)  : 270, 283
     .   : milestone, 277,

    section CallTarget+Inlining+NGEN
    This PR (4683) - mean (1,101ms)  : 1071, 1131
     .   : milestone, 1101,
    master - mean (1,101ms)  : 1074, 1129
     .   : milestone, 1101,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4683) - mean (262ms)  : 256, 267
     .   : milestone, 262,
    master - mean (267ms)  : 260, 273
     .   : milestone, 267,

    section CallTarget+Inlining+NGEN
    This PR (4683) - mean (1,043ms)  : 1016, 1071
     .   : milestone, 1043,
    master - mean (1,055ms)  : 1027, 1082
     .   : milestone, 1055,

```

File: profiler/src/Demos/WcfService/Host/WcfServiceHost.csproj [hunk 1]
```diff
@@ -53,10 +54,6 @@
       <Project>{b0882e91-443b-4161-ab64-07e188067d3f}</Project>
       <Name>Datadog.Demos.Util</Name>
     </ProjectReference>
-    <ProjectReference Include="..\Library\WcfServiceLibrary.csproj">
```
Comment (chrisnas) [inline profiler/src/Demos/WcfService/Host/WcfServiceHost.csproj:56]: It was not needed any more?
Check that it is still compiling locally

**Review by chrisnas — APPROVED** [review]
LGTM 
just a couple of comments

File: profiler/src/Demos/Shared/Util/Datadog.Demos.Util.csproj [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
```
[the remark this answers is not in the document]
Comment (gleocadie) [reply profiler/src/Demos/Shared/Util/Datadog.Demos.Util.csproj:3]: 👍

File: profiler/src/Demos/WcfService/Host/WcfServiceHost.csproj (hunk 1 above)
Comment (gleocadie) [reply profiler/src/Demos/WcfService/Host/WcfServiceHost.csproj:56]: 👍

**Comment by andrewlock** [comment]
## Throughput/Crank Report:zap:

Throughput results for AspNetCoreSimpleController comparing the following branches/commits:
- [This PR (4683)](https://github.com/DataDog/dd-trace-dotnet/tree/336bce2d618fe424a86e4a46b7bd6db7ff385b85)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/8cb9a9a49527481d07028f07199fbfb3b6e19cf9)
- [benchmarks/2.9.0](https://github.com/DataDog/dd-trace-dotnet/tree/89ae3f4ee7d85b824badda5873166707c9e560b4)

Cases where throughput results for the PR are worse than latest master (5% drop or greater), results are shown in **red**.

Note that these results are based on a _single_ point-in-time result for each branch. For full results, see [one](https://ddstaging.datadoghq.com/dashboard/fnz-afi-c2i/apm-net-crank) of the [many](https://ddstaging.datadoghq.com/dashboard/c5g-i7d-kcy/ciapp-apm-net-throughput-tests), [many](https://ddstaging.datadoghq.com/dashboard/uxh-m8j-qhi/ciapp-apm-net-throughput-tests-kevin) dashboards!

```mermaid
gantt
    title Throughput Linux x64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (4683) (11.675M)   : 0, 11674586
    master (11.841M)   : 0, 11841324
    benchmarks/2.9.0 (11.683M)   : 0, 11682730

    section Automatic
    This PR (4683) (8.100M)   : 0, 8100382
    master (8.114M)   : 0, 8113881
    benchmarks/2.9.0 (8.466M)   : 0, 8466123

    section Trace stats
    This PR (4683) (8.455M)   : 0, 8454792
    master (8.431M)   : 0, 8430732

    section Manual
    This PR (4683) (10.222M)   : 0, 10222481
    master (10.308M)   : 0, 10307770

    section Manual + Automatic
    This PR (4683) (15.700M)   : 0, 15700388
    master (7.667M)   : 0, 7667050

    section Version Conflict
    This PR (4683) (6.958M)   : 0, 6958329
    master (6.964M)   : 0, 6964378

```
```mermaid
gantt
    title Throughput Linux arm64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (4683) (9.599M)   : 0, 9598783
    master (9.654M)   : 0, 9654151
    benchmarks/2.9.0 (9.501M)   : 0, 9501348

    section Automatic
    This PR (4683) (6.660M)   : 0, 6660317
    master (6.684M)   : 0, 6684226

    section Trace stats
    This PR (4683) (6.921M)   : 0, 6920884
    master (6.933M)   : 0, 6932507

    section Manual
    This PR (4683) (8.184M)   : 0, 8184176
    master (8.283M)   : 0, 8283486

    section Manual + Automatic
    This PR (4683) (6.184M)   : 0, 6184144
    master (6.220M)   : 0, 6219934

    section Version Conflict
    This PR (4683) (5.723M)   : 0, 5722835
    master (5.717M)   : 0, 5716509

```
```mermaid
gantt
    title Throughput Windows x64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (4683) (10.273M)   : 0, 10273085
    master (10.163M)   : 0, 10163101
    benchmarks/2.9.0 (10.646M)   : 0, 10646425

    section Automatic
    This PR (4683) (7.365M)   : 0, 7365441
    master (7.381M)   : 0, 7380573
    benchmarks/2.9.0 (7.738M)   : 0, 7737732

    section Trace stats
    This PR (4683) (7.692M)   : 0, 7691736
    master (7.665M)   : 0, 7664749

    section Manual
    This PR (4683) (9.002M)   : 0, 9001952
    master (9.073M)   : 0, 9072707

    section Manual + Automatic
    This PR (4683) (7.046M)   : 0, 7046443
    master (7.046M)   : 0, 7046401

    section Version Conflict
    This PR (4683) (6.294M)   : 0, 6293530
    master (6.456M)   : 0, 6455711

```
```mermaid
gantt
    title Throughput Linux x64 (ASM) (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    master (7.335M)   : 0, 7335358
    benchmarks/2.9.0 (7.766M)   : 0, 7765742

    section No attack
    master (1.964M)   : 0, 1964432
    benchmarks/2.9.0 (3.225M)   : 0, 3225290

    section Attack
    master (1.574M)   : 0, 1573620
    benchmarks/2.9.0 (2.447M)   : 0, 2447478

    section Blocking
    master (3.216M)   : 0, 3215907

    section IAST default
    master (6.392M)   : 0, 6391939

    section IAST full
    master (5.708M)   : 0, 5708333

    section Base vuln
    master (0.963M)   : 0, 962637

    section IAST vuln
    master (0.859M)   : 0, 859241

```

**Pull request merged**